stream/src/main/scala/org/apache/pekko/stream/javadsl/Graph.scala

```diff
@@ -742,7 +742,7 @@ object GraphDSL extends GraphCreate {
     new GenericGraph(s, gbuilder.delegate.result(s))
   }
 
-  final class Builder[+Mat]()(private[stream] implicit val delegate: scaladsl.GraphDSL.Builder[Mat]) { self =>
+  final class Builder[+Mat](private[stream] implicit val delegate: scaladsl.GraphDSL.Builder[Mat]) { self =>
```

Comment:
How do implicit args work when calling from Java? Is it possible that this constructor is not meant to be used directly by Java users, that is for use by internal Pekko code?

Reply:
implicit args also get flattened to flat parameter lists since under the hood they are also implemented via currying (with only Scala having the ability to pass those arguments implicitly).

Reply:
But do you think that this constructor is to meant to be called by external Java users? It looks like an internal Pekko use only constructor.

Reply:
On the surface you seem to be right although it is referred to in multiple places in the docs.

In any case even if its only meant for internal Pekko usage, this change is harmless.

Reply:
This change is fine, but it does break some binary compatibility, I'm using Javadsl only in Java, so I think that should also apply to others.

Reply:
If it breaks binary compatibility MiMa would detect it